internal/dataplane/kong_client.go

```diff
@@ -41,6 +41,10 @@ const (
 	KongConfigurationApplyFailedEventReason = "KongConfigurationApplyFailed"
 )
 
+type ClientsProvider interface {
```

Comment:
Since we're not in the Admin API related package this might better include adminapi in the name, WDYT? Client is a bit vague in here I'd say.

If we'd like to abstract away admin api out of this (which is also an option allowing other dataplanes to be configured if we ever desire to 🤷 ) then the return value from `Clients()` should also have its own interface, i.e. don't return `[]*adminapi.Client` but an interface.

Having said that I'd maybe sprinkle some comments on this interface and add aforementioned adminapi prefix or something like that to the name.

Reply:
Yeah, good point. 👍 I added the prefix and a comment to explain what the interface is meant for.